patroni/config_generator.py

```diff
@@ -268,7 +268,8 @@ def generate(self) -> None:
         wal_level = 'hot_standby' if self.pg_major < 90600 else 'replica'
         self.config['bootstrap']['dcs']['postgresql']['parameters']['wal_level'] = wal_level
 
-        self.config['bootstrap']['dcs']['postgresql']['use_pg_rewind'] = True
+        self.config['bootstrap']['dcs']['postgresql']['use_pg_rewind'] = True \
+            if self.config['bootstrap']['dcs']['postgresql']['parameters']['wal_log_hints'] == "on" else False
```

Comment:
```suggestion
        self.config['bootstrap']['dcs']['postgresql']['use_pg_rewind'] = \
            parse_bool(self.config['bootstrap']['dcs']['postgresql']['parameters']['wal_log_hints']) is True
```
But, TBH there is no need to do that, because we are not changing the default value of `wal_log_hints`.

Reply:
Do you want me to add the following contents as comments?

- "Either the cluster must be initialized with Data page checksums (--data-checksums option for initdb) and/or wal_log_hints must be set to on, or pg_rewind will not work."

I will commit an example for your request below. Check plz.

Reply:
I will fix it 👍